connect/json/src/test/java/org/apache/kafka/connect/json/JsonConverterTest.java

```diff
@@ -791,4 +798,16 @@ private void validateEnvelopeNullSchema(JsonNode env) {
         assertTrue(env.get(JsonSchema.ENVELOPE_SCHEMA_FIELD_NAME).isNull());
         assertTrue(env.has(JsonSchema.ENVELOPE_PAYLOAD_FIELD_NAME));
     }
+    
+    private boolean convertStructEqual(Schema schema, Struct struct) {
+        try {
+            converter.fromConnectData(TOPIC, schema, struct);
+        } catch (DataException exc) {
+            if (!exc.getMessage().contains("serialization")) {
+                return false;
+            }
+            return true;
+        }
+        return true;
```

Comment:
This method does check the logic in the JSON converter, but it'd be good to also explicitly add a check here as well just to ensure that logic is sound:

    assertEquals(schema, struct.schema());